internal/cli/root/builder.go

```diff
@@ -139,18 +144,49 @@ func (opts *BuilderOpts) hasNewVersionAvailable() (newVersionAvailable bool, new
 		return false, "", err
 	}
 
-	svLatestVersion, err := semver.NewVersion(latestVersion)
+	svLatestVersion, err := semver.NewVersion(latestVersion.Version)
 	if err != nil {
 		return false, "", err
 	}
 
-	if svCurrentVersion.Compare(svLatestVersion) < 0 {
-		return true, latestVersion, nil
+	if svCurrentVersion.Compare(svLatestVersion) < 0 && (!isHomebrew() || isAtLeast24HoursPast(latestVersion.PublishedAt)) {
+		return true, latestVersion.Version, nil
 	}
 
 	return false, "", nil
 }
 
+func isAtLeast24HoursPast(t time.Time) bool {
+	return !t.IsZero() && time.Since(t) >= time.Hour*24
+}
+
+func isHomebrew() bool {
+	if runtime.GOOS != "darwin" {
+		return false
+	}
+
+	brewPath, err := exec.LookPath("brew")
+	if err != nil {
+		return false
+	}
+
+	brewPrefixBytes, err := exec.Command(brewPath, "--prefix").Output()
+	if err != nil {
+		return false
+	}
+
+	brewPrefixPath := strings.TrimSpace(string(brewPrefixBytes))
+	brewBinPrefix := filepath.Join(brewPrefixPath, "bin") + string(filepath.Separator)
+
+	mongocliPath, err := os.Executable()
+
+	if err != nil {
+		return false
+	}
+
+	return strings.HasPrefix(mongocliPath, brewBinPrefix)
```

Comment:
[q] wouldn't you be able to use only `brewPrefixPath` here? asking if we can avoid the appending of `bin`